docs/source/notes/gradcheck.rst

```diff
@@ -0,0 +1,218 @@
+.. _gradcheck-mechanics:
+
+Gradcheck mechanics
+===================
+
+This note presents an overview of how the :meth:`~torch.autograd.gradcheck` and :meth:`~torch.autograd.gradgradcheck` functions work.
+
+It will cover both forward and backward mode AD as well as complex valued functions and higher order derivatives.
```

Comment:
```suggestion
It will cover both forward and backward mode AD for both real and complex valued functions as well as and higher order derivatives.
```